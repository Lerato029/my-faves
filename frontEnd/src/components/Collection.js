/* =================================================================The Collection Component======================================================= */
/* import statement to be able to create React component */
import React from "react";

//import bootstrap button
import { Button } from "react-bootstrap";

export default function Collection(props) {
  //destructuring props for better legibility
  let { collectionItems, removeItem } = props;

  //return statement where each entry in the collectionItems array is given the format below
  return (
    <div class="row">
      {collectionItems.map((collection, index) => (
        <div class="column" key={index}>
          <div class="card">
            <img src={collection.artworkUrl100} alt={collection.trackName} />

            <h5 className="blackFont">{collection.trackName}</h5>
            <p className="blackFont">{collection.kind}</p>
            <Button
              variant="danger"
              className="cardBtn"
              onClick={() => {
                removeItem(index);
              }}
            >
              DELETE
            </Button>
          </div>
        </div>
      ))}
    </div>
  );
}
