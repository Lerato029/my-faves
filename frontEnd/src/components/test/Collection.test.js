/* =================================================================The App Component======================================================= */ import React from "react";
import renderer from "react-test-renderer";
import Collection from "../Collection";

it("renders correctly", () => {
  const collectionItems = [
    {
      artworkUrl100: "",
      trackName: "Life of a bee",
      kind: "Movie",
    },
  ];

  const removeItem = () => {
    console.log("removeItem called");
  };

  const tree = renderer
    .create(
      <Collection collectionItems={collectionItems} removeItem={removeItem} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
