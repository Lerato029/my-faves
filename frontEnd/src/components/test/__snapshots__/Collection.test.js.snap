// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="row"
>
  <div
    class="column"
  >
    <div
      class="card"
    >
      <img
        alt="Life of a bee"
        src=""
      />
      <h5
        className="blackFont"
      >
        Life of a bee
      </h5>
      <p
        className="blackFont"
      >
        Movie
      </p>
      <button
        className="btn btn-danger"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        DELETE
      </button>
    </div>
  </div>
</div>
`;
